### PR TITLE
debuginfo: Make sure uploads do not block

### PR DESCRIPTION
**Pull request by kakkoyun**
Signed-off-by: Kemal Akkoyun <kakkoyun@gmail.com>

Supersedes https://github.com/parca-dev/parca-agent/pull/1038

### CODE REVIEW

File: pkg/debuginfo/manager.go [hunk 1]
```diff
@@ -123,25 +128,30 @@ func (di *Manager) removeAsUploading(buildID string) {
...
-			g.Go(func() error {
-				err := di.ensureUploaded(ctx, objFile)
-				if err != nil {
-					level.Error(logger).Log("msg", "failed to ensure debuginfo is uploaded", "err", err)
-				}
-				return nil
-			})
+	go func() {
+		_, err, _ := di.sfg.Do("ensure-uploaded", func() (interface{}, error) {
+			g := errgroup.Group{} // errgroup.WithContext doesn't work for this use case, we want to continue uploading even if one fails.
```
Comment (javierhonduco) [inline pkg/debuginfo/manager.go:133]: Should we use [`WithContext`](https://cs.opensource.google/go/x/sync/+/refs/tags/v0.1.0:errgroup/errgroup.go;l=45;bpv=1;bpt=1)?

**Review by javierhonduco — APPROVED** [review]

File: pkg/debuginfo/manager.go (hunk 1 above)
Comment (kakkoyun) [reply pkg/debuginfo/manager.go:133]: I have added the comment for this question 😅

Comment (kakkoyun) [reply pkg/debuginfo/manager.go:133]: But if the context gets canceled from the caller, we already propagate it down, so the grpc calls will be canceled. And group will exit.

Comment (javierhonduco) [reply pkg/debuginfo/manager.go:133]: > I have added the comment for this question 😅

Oops, my bad, didn't see it

**Pull request merged**